torch/csrc/jit/passes/lower_graph.cpp

```diff
@@ -114,16 +116,30 @@ std::pair<std::shared_ptr<Graph>, std::vector<Slot>> lower_graph(
   return std::make_pair(std::move(g), std::move(extra_ivalues));
 }
 
-static std::vector<at::Tensor> loadTensors(const std::vector<Slot>& slots) {
-  std::vector<at::Tensor> result;
+static std::vector<IValue> loadTensors(const std::vector<Slot>& slots) {
+  std::vector<IValue> result;
   result.reserve(slots.size());
   for (const Slot& slot : slots) {
-    result.emplace_back(slot.obj->getSlot(slot.offset).toTensor());
+    auto obj = slot.obj->getSlot(slot.offset);
+    if (obj.isTensor()) {
+      result.emplace_back(obj.toTensor());
+    } else {
+      // Unpack quantization packed tensor
+      auto type = obj.type();
+      TORCH_CHECK(
+          type ==
+              getCustomClass("__torch__.torch.classes.LinearPackedParamsBase"),
+          "Unknown type ",
+          type->python_str(),
+          " encountered in graph lowering. This type is not supported in ONNX export.");
+      result.emplace_back(
+          script::Object(obj.toObject()).run_method("__getstate__"));
```

Comment:
just curious - why do we need to call _getstate_ here instead of keeping the class as is and invoking `__getstate__` in unpackQuantizedWeightsHelper (so that there's one less magic place)

Reply:
question still stands